components/script/dom/formdataevent.rs

```diff
@@ -0,0 +1,97 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use crate::dom::bindings::codegen::Bindings::EventBinding::EventMethods;
+use crate::dom::bindings::codegen::Bindings::FormDataEventBinding;
+use crate::dom::bindings::codegen::Bindings::FormDataEventBinding::FormDataEventMethods;
+use crate::dom::bindings::error::{Error, Fallible};
+use crate::dom::bindings::inheritance::Castable;
+use crate::dom::bindings::reflector::{reflect_dom_object, DomObject};
+use crate::dom::bindings::root::{Dom, DomRoot};
+use crate::dom::bindings::str::DOMString;
+use crate::dom::event::Event;
+use crate::dom::event::{EventBubbles, EventCancelable};
+use crate::dom::formdata::FormData;
+use crate::dom::globalscope::GlobalScope;
+use crate::dom::window::Window;
+use dom_struct::dom_struct;
+use servo_atoms::Atom;
+
+#[dom_struct]
+pub struct FormDataEvent {
+    event: Event,
+    form_data: Dom<FormData>,
+}
+
+impl FormDataEvent {
+    fn new_inherited(form_data: &FormData) -> FormDataEvent {
+        FormDataEvent {
+            event: Event::new_inherited(),
+            form_data: Dom::from_ref(form_data),
+        }
+    }
+
+    pub fn new_uninitialized(global: &GlobalScope, form_data: &FormData) -> DomRoot<FormDataEvent> {
```

Comment:
No need for this function; there are no callers so we can inline it in `new` instead.